homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -149,98 +95,212 @@ def name(self):
+                partial(RemoteControl, self._host, self._port, **params)
+            )
+
+            self.state = STATE_ON
+            self.available = True
+        except (TimeoutError, URLError, OSError) as err:
+            if control_existed or during_setup:
+                _LOGGER.error("Could not establish remote connection: %s", err)
+                self._control = None
+                self.state = STATE_OFF
+                self.available = False
+        except Exception as err:  # pylint: disable=broad-except
+            if control_existed or during_setup:
+                _LOGGER.error("An unknown error occurred: %s", err)
+                self._control = None
+                self.state = STATE_OFF
+                self.available = False
+
+    async def async_update(self):
+        """Retrieve the latest data."""
+        if self._control is None:
+            await self.async_create_remote_control()
+            return
+
+        try:
+            self.muted = self._control.get_mute()
+            self.volume = self._control.get_volume() / 100
+
+            self.state = STATE_ON
+            self.available = True
+        except EncryptionRequired:
+            _LOGGER.error("The connection couldn't be encrypted")
+        except (TimeoutError, URLError, OSError):
+            self.state = STATE_OFF
+            self.available = False
+            await self.async_create_remote_control()
+
+    async def async_send_key(self, key):
+        """Send a key to the TV and handles exceptions."""
+        try:
+            self._control.send_key(key)
+        except EncryptionRequired:
+            _LOGGER.error("The connection couldn't be encrypted")
+        except (TimeoutError, URLError, OSError):
+            self.state = STATE_OFF
+            self.available = False
+            await self.async_create_remote_control()
+
+    async def async_set_mute(self, enable):
+        """Set mute based on 'enable'."""
+        try:
+            self._control.set_mute(enable)
+        except EncryptionRequired:
+            _LOGGER.error("The connection couldn't be encrypted")
+        except (TimeoutError, URLError, OSError):
+            self.state = STATE_OFF
+            self.available = False
+            await self.async_create_remote_control()
+
+    async def async_set_volume(self, volume):
+        """Set volume level, range 0..1."""
+        volume = int(volume * 100)
+        try:
+            self._control.set_volume(volume)
+        except EncryptionRequired:
+            _LOGGER.error("The connection couldn't be encrypted")
+        except (TimeoutError, URLError, OSError):
+            self.state = STATE_OFF
+            self.available = False
+            await self.async_create_remote_control()
+
+    async def async_play_media(self, media_type, media_id):
         """Play media."""
         _LOGGER.debug("Play media: %s (%s)", media_id, media_type)
 
         if media_type == MEDIA_TYPE_URL:
```

Comment:
Invert this check and if true log the warning and return. Then we can outdent below.